kernel/kernel-api/src/main/java/io/delta/kernel/internal/actions/CommitInfo.java

```diff
@@ -148,4 +149,19 @@ public Row toRow() {
 
         return new GenericRow(CommitInfo.FULL_SCHEMA, commitInfo);
     }
+
+    /**
+     * Returns the `inCommitTimestamp` of the given `commitInfoOpt` if it is defined.
+     * Throws an exception if `commitInfoOpt` is empty or contains an empty `inCommitTimestamp`.
+     */
+    public static long getRequiredInCommitTimestamp(
+            Optional<CommitInfo> commitInfoOpt, String version) {
```

Comment:
I think we can just throw InvalidTableException when loading the commitInfo (if not found). That avoids the one of the checks we have here.